# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLTransformFeedback.cpp
# Commit: ccfa767dba64
# Full Hash: ccfa767dba644dc193e0246eb8e8ff3377e8b8a5
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Client-side bindings mirror for precise CC, and merge similar codepaths. r=handyman
#   
#   * Context loss using RAII
#   * Move Program reflection Client-side
#   
# ==============================================================================

diff -r 71c122ac0ca7 -r ccfa767dba64 dom/canvas/WebGLTransformFeedback.cpp
--- a/dom/canvas/WebGLTransformFeedback.cpp	Wed Jan 08 22:19:14 2020 +0000
+++ b/dom/canvas/WebGLTransformFeedback.cpp	Wed Jan 08 22:19:16 2020 +0000
@@ -14,21 +14,17 @@
 namespace mozilla {
 
 WebGLTransformFeedback::WebGLTransformFeedback(WebGLContext* webgl, GLuint tf)
-    : WebGLRefCountedObject(webgl),
+    : WebGLContextBoundObject(webgl),
       mGLName(tf),
-      mIndexedBindings(webgl->mGLMaxTransformFeedbackSeparateAttribs),
+      mIndexedBindings(webgl->Limits().maxTransformFeedbackSeparateAttribs),
       mIsPaused(false),
-      mIsActive(false) {
-  mContext->mTransformFeedbacks.insertBack(this);
-}
+      mIsActive(false) {}
 
-WebGLTransformFeedback::~WebGLTransformFeedback() { DeleteOnce(); }
-
-void WebGLTransformFeedback::Delete() {
+WebGLTransformFeedback::~WebGLTransformFeedback() {
+  if (!mContext) return;
   if (mGLName) {
     mContext->gl->fDeleteTransformFeedbacks(1, &mGLName);
   }
-  removeFrom(mContext->mTransformFeedbacks);
 }
 
 ////////////////////////////////////////